p2p/dialer.go

```diff
@@ -110,6 +114,15 @@ func (m *dialer) exchangeConfig(consumerID, providerID identity.Identity, servic
 	}
 	defer brokerConn.Close()
 
+	ready := make(chan bool)
+	defer close(ready)
+	_, err = brokerConn.Subscribe(channelHandlersReadySubject(providerID, serviceType), func(msg *nats_lib.Msg) {
+		if err := m.channelHandlersReady(msg, ready); err != nil {
```

Comment:
If you put this defer inside the function, you don't need to pass channel to the `channelHandlersReady`, it will be just closed when `channelHandlersReady` is done.
```suggestion
	_, err = brokerConn.Subscribe(channelHandlersReadySubject(providerID, serviceType), func(msg *nats_lib.Msg) {
		defer close(ready)
		if err := m.channelHandlersReady(msg); err != nil {
```